test/widgets_test.rb

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+require "users/widgets"
+
+def stub_widget_value(id, value)
+  allow(Yast::UI).to receive(:QueryWidget).with(Id(id), :Value).and_return(value)
+end
+
+describe Users::PasswordWidget do
+  it "has help text" do
+    expect(subject.help).to_not be_empty
+  end
+
+  it "has valid content" do
+    expect(subject.contents).to be_a(Yast::Term)
+  end
+
+  it "initializes password to current value" do
+    pwd = "paranoic"
+    allow(Yast::UsersSimple).to receive(:GetRootPassword).and_return(pwd)
+    expect(Yast::UI).to receive(:ChangeWidget).with(Id(:pw1), :Value, pwd)
+    expect(Yast::UI).to receive(:ChangeWidget).with(Id(:pw2), :Value, pwd)
+
+    subject.init
+  end
+
+  context "validation" do
+    it "reports error if password is empty" do
+      stub_widget_value(:pw1, "")
+      stub_widget_value(:pw2, "")
+
+      expect(Yast::Popup).to receive(:Error)
+      expect(Yast::UI).to receive(:SetFocus).with(Id(:pw1))
+
+      expect(subject.validate).to eq false
+    end
+
+    it "reports error if passwords do not match" do
+      stub_widget_value(:pw1, "a")
+      stub_widget_value(:pw2, "b")
+
+      expect(Yast::Popup).to receive(:Message)
+      expect(Yast::UI).to receive(:SetFocus).with(Id(:pw2))
+
+      expect(subject.validate).to eq false
+    end
+
+    it "reports error if password contain forbidden characters" do
+      stub_widget_value(:pw1, "mimic_forbidden")
+      stub_widget_value(:pw2, "mimic_forbidden")
+
+      expect(Yast::UsersSimple).to receive(:CheckPassword).with("mimic_forbidden", "local").
+        and_return("Invalid password")
+      expect(Yast::Report).to receive(:Error)
+      expect(Yast::UI).to receive(:SetFocus).with(Id(:pw1))
+
+      expect(subject.validate).to eq false
+    end
+
+    it "asks for confirmation if password is not strong enough" do
+      stub_widget_value(:pw1, "a")
```

Comment:
NP: This test is a bit tricky, at the first sight it might seem that after setting the password to something like `eh3Roh5iefur2eew` the test should fail. But the validation is actually mocked, the value here is ignored.

I'd change the title to something some generic, like `asks for confirmation if password validation fails`.